contracts/ServiceNodeRewards.sol

```diff
@@ -229,22 +238,41 @@ contract ServiceNodeRewards is Initializable, Ownable2StepUpgradeable, PausableU
     }
 
     /// @dev Internal function to handle reward claims. Will transfer the
-    /// available rewards worth of our token to claimingAddress
+    /// requested amount of our token to claimingAddress, up to the available rewards
     /// @param claimingAddress The address claiming the rewards.
-    function _claimRewards(address claimingAddress) internal {
+    /// @param amount The amount of rewards to claim.
+    function _claimRewards(address claimingAddress, uint256 amount) internal {
         uint256 claimedRewards = recipients[claimingAddress].claimed;
         uint256 totalRewards = recipients[claimingAddress].rewards;
-        uint256 amountToRedeem = totalRewards - claimedRewards;
-
-        recipients[claimingAddress].claimed = totalRewards;
-        emit RewardsClaimed(claimingAddress, amountToRedeem);
+        uint256 maxAmount = totalRewards - claimedRewards;
+        if (amount > maxAmount)
+            revert MaxClaimExceeded();
+
+        uint256 _epochDay = block.timestamp / 86400;
+        if (_epochDay > epochDay) {
+            epochDay = _epochDay;
+            periodicClaims = 0;
+        }
+        periodicClaims += amount;
+        if (periodicClaims > claimThreshold) revert ClaimThresholdExceeded();
 
-        SafeERC20.safeTransfer(designatedToken, claimingAddress, amountToRedeem);
+        recipients[claimingAddress].claimed += amount;
+        emit RewardsClaimed(claimingAddress, amount);
+        SafeERC20.safeTransfer(designatedToken, claimingAddress, amount);
     }
 
-    /// @notice Claim the rewards due for the active wallet invoking the claim.
+    /// @notice Claim all available rewards for the active wallet invoking the claim.
     function claimRewards() public {
-        _claimRewards(msg.sender);
+        uint256 claimedRewards = recipients[msg.sender].claimed;
+        uint256 totalRewards = recipients[msg.sender].rewards;
+        uint256 amountToRedeem = totalRewards - claimedRewards;
+        _claimRewards(msg.sender, amountToRedeem);
```

Comment:
One potential problem: suppose I have 1.1M in my unclaimed balance, then this is always going to be fail and my 1.1 will be stuck.

I think perhaps we can update this so that amountToRedeem is the smaller of the available periodic amount, and my claimed amount, and so in such a scenario I would get 700000 (or whatever is available) SENT out the first time I call it, then I have to wait until the next period to get the remaining 400000 by calling it again.

Reply:
Never mind, there's a way out through the following version taking an amount that I missed.